applications/jonny5/src/j5_balcraw_worker.erl

```diff
@@ -0,0 +1,118 @@
+%%%-------------------------------------------------------------------
+%%% File    : j5_balacne_crawler.erl
+%%% Description : Jonny5 module for disconnect calls when account
+%%% balance drops below zero
+%%%-------------------------------------------------------------------
+-module(j5_balcraw_worker).
+
+-export([start/0]).
+
+-export([maybe_disconnect_account/1]).
+-export([disconnect_account/1]).
+
+-include("jonny5.hrl").
+
+-define(SERVER, ?MODULE).
+-define(DEFAULT_INTERACCOUNT_DELAY, 10).
+
+-spec start() -> no_return().
+start() ->
+    case j5_channels:accounts() of
+        [] -> exit("no accounts");
+        Accounts ->
+            lager:debug("check ~p account(s) for zero balance", [length(Accounts)]),
+            crawler(Accounts)
+    end.
+
+-spec crawler(api_binaries()) -> no_return().
+crawler([]) -> exit("work done");
+crawler([Account|Accounts]) ->
+    maybe_disconnect_account(Account),
+    Delay = whapps_config:get_integer(?APP_NAME, <<"balance_crawler_interaccount_delay_ms">>, ?DEFAULT_INTERACCOUNT_DELAY),
+    timer:sleep(Delay),
+    crawler(Accounts).
+
+-spec maybe_disconnect_account(ne_binary()) -> 'ok'.
+maybe_disconnect_account(AccountId) ->
+    Limits = j5_limits:get(AccountId),
+    DisconnectActiveCalls = j5_limits:disconnect_active_calls(Limits),
+    case wh_util:is_true(DisconnectActiveCalls) of
+        'false' -> 'ok';
+        'true' ->
+            case j5_per_minute:maybe_credit_available(0, Limits, 'true') of
+                'true' -> 'ok';
+                'false' -> disconnect_account(AccountId)
+            end
+    end.
+
+-spec disconnect_account(ne_binary()) -> 'ok'.
+disconnect_account(AccountId) ->
+    case j5_channels:per_minute(AccountId) of
+        Count when Count > 0 ->
+            lager:debug("account ~p has ~p per-minute calls, disconnect them",[AccountId, Count]),
+            maybe_disconnect_channels(j5_channels:account(AccountId));
+        _ ->
+            lager:debug("account ~p doesn't have any per-minute call",[AccountId]),
+            'ok'
+    end.
+
+-spec maybe_disconnect_channels(j5_channels:channels()) -> 'ok'.
+maybe_disconnect_channels([]) -> 'ok';
+maybe_disconnect_channels([Channel|Channels]) ->
+    Props = j5_channels:to_props(Channel),
+    case props:get_binary_value(<<"Account-Billing">>, Props) =:= <<"per_minute">>
+         orelse props:get_binary_value(<<"Reseller-Billing">>, Props) =:= <<"per_minute">>
```

Comment:
`=:=` is only useful if one wants to compare floats with high precision. Use `==`.